coderd/aitasks.go

```diff
@@ -302,6 +302,35 @@ func taskFromDBTaskAndWorkspace(dbTask database.Task, ws codersdk.Workspace) cod
 		}
 	}
 
+	// If no valid agent state was found for the current build and the task is initializing,
+	// provide a descriptive initialization message.
+	if currentState == nil && codersdk.TaskStatus(dbTask.Status) == codersdk.TaskStatusInitializing {
+		message := "Initializing workspace"
+
+		switch {
+		case ws.LatestBuild.Status == codersdk.WorkspaceStatusPending:
+			message = "Workspace build is pending"
+		case ws.LatestBuild.Status == codersdk.WorkspaceStatusStarting:
+			message = "Starting workspace"
+		case taskAgentLifecycle != nil:
+			switch *taskAgentLifecycle {
+			case codersdk.WorkspaceAgentLifecycleCreated:
+				message = "Agent is connecting"
```

Comment:
We could also probably generate this using the workspace agent lifecycle status?